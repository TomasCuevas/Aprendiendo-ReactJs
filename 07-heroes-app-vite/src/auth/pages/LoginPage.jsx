import { AuthContext } from '../context/AuthContext';
import { useContext } from 'react';
import { useNavigate } from 'react-router-dom';

export const LoginPage = () => {
  const navigate = useNavigate();
  const { login } = useContext( AuthContext );
  
  const handleLogin = () => {
    const lasthPath = JSON.parse( localStorage.getItem('lastPath') ) || '/marvel';
    
    login( 'Tom√°s Cuevas' );
    navigate(lasthPath, {
      replace: true
    });
  }

  return (
    <div className="container mt-5">
      <h1>LoginPage</h1>
      <hr />

      <button 
        className="btn btn-primary"
        onClick={ handleLogin }
      >
        Login
      </button>
    </div>
  )
}