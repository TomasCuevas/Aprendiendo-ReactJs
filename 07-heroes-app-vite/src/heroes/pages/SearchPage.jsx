import { useNavigate, useLocation } from 'react-router-dom';
import queryString from 'query-string';
import { useForm } from '../../hooks/useForm';
import { HeroCard } from '../components';
import { getHeroesByName } from '../helpers';

export const SearchPage = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const { q = '' } = queryString.parse( location.search )
  const heroes = getHeroesByName( q );
  
  const { searchText, onInputChange } = useForm({
    searchText: q
  })

  const onSearchSubmit = ( e ) => {
    e.preventDefault();

    navigate(`?q=${searchText}`);
  } 

  return (
    <>
      <h1>Search</h1>
      <hr />

      <div className="row">
        <div className="col-5 mt-5">
          <h4>Searching</h4>
          <hr />

          <form onSubmit={ onSearchSubmit }>
            <input 
              aria-label="search-input"
              type="text" 
              placeholder="Search a hero"
              className="form-control"
              name="searchText"
              autoComplete="off"
              value={ searchText }
              onChange={ onInputChange }
            />

            <button 
              aria-label="search-button"
              type="submit"
              className="btn btn-outline-primary mt-2"
            >
              Search
            </button>
          </form>
        </div>

        <div className="col-7 mt-5">
          <h4>Result</h4>
          <hr />

          {
            (q.length <= 1) && (
              <div className="alert alert-primary">
                Search a hero
              </div>
            )
          }

          {
            (q.length > 0 && heroes.length === 0) && (
              <div className="alert alert-danger">
                Not hero with <b>{ q }</b>
              </div>
            )
          }

          {
            ( heroes.length > 0 ) && (
              heroes.map((hero) => (
                <HeroCard key={ hero.id } { ...hero } />
              ))
            )
          }

        </div>
      </div>

    </>
  )
}