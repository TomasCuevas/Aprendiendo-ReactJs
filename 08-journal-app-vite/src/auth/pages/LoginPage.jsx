/* eslint-disable no-extra-boolean-cast */
import { useMemo } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link as Routerlink } from 'react-router-dom';
import { Grid, Typography, TextField, Button, Link, Alert } from '@mui/material';
import { Google } from '@mui/icons-material';

import { AuthLayout } from '../layout/AuthLayout';

import { useForm } from '../../hooks/useForm';
import { startLoginWithEmailPassword, startGoogleSignin } from '../../store/auth/thunks';

const formValues = {
  email: '',
  password: '',
};

export const LoginPage = () => {
  const dispatch = useDispatch();
  const { status, errorMessage } = useSelector((state) => state.auth);
  const { formState, email, password, onInputChange } = useForm(formValues);

  const isAuthenticating = useMemo(() => status === 'checking', [status]);

  const onSubmit = (event) => {
    event.preventDefault();
    dispatch(startLoginWithEmailPassword(formState));
  };

  const onGoogleSignin = () => {
    dispatch(startGoogleSignin());
  };

  return (
    <AuthLayout formTitle="Login">
      <form onSubmit={onSubmit}>
        <Grid container>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              label="Correo"
              type="email"
              placeholder="correo@google.com"
              fullWidth
              value={email}
              name="email"
              onChange={onInputChange}
            />
          </Grid>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              label="Contraseña"
              type="password"
              placeholder="contraseña"
              fullWidth
              value={password}
              name="password"
              onChange={onInputChange}
            />
          </Grid>
        </Grid>

        <Grid container spacing={2} sx={{ mb: 2, mt: 1 }}>
          <Grid item xs={12} display={!!errorMessage ? 'block' : 'none'}>
            <Alert severity="error">{errorMessage}</Alert>
          </Grid>

          <Grid item xs={12} sm={6}>
            <Button disabled={isAuthenticating} type="submit" variant="contained" fullWidth>
              Login
            </Button>
          </Grid>

          <Grid item xs={12} sm={6}>
            <Button
              disabled={isAuthenticating}
              onClick={onGoogleSignin}
              variant="contained"
              fullWidth
              aria-label="google-btn"
            >
              <Google />
              <Typography sx={{ ml: 1 }}>Google</Typography>
            </Button>
          </Grid>
        </Grid>

        <Grid container direction="row" justifyContent="end">
          <Link component={Routerlink} to="/auth/register">
            Crear una cuenta
          </Link>
        </Grid>
      </form>
    </AuthLayout>
  );
};
