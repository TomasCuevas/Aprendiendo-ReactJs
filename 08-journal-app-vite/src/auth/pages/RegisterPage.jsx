import { useState } from "react";
import { Link as Routerlink } from "react-router-dom";
import { Grid, Typography, TextField, Button, Link } from "@mui/material";

import { useForm } from "../../hooks/useForm";
import { AuthLayout } from "../layout/AuthLayout";

const formValues = {
  displayName: "",
  email: "",
  password: "",
};

const formValidations = {
  email: [(value) => value.includes("@"), 'El correo debe de tener un "@"'],
  password: [
    (value) => value.length >= 6,
    "El password debe tener minimo 6 caracteres.",
  ],
  displayName: [
    (value) => value.length >= 2,
    "El nombre debe tener minimo 2 caracteres.",
  ],
};

export const RegisterPage = () => {
  const [formSubmitted, setFormSubmitted] = useState(false);

  const {
    displayName,
    displayNameValid,
    email,
    emailValid,
    formState,
    isFormValid,
    onInputChange,
    password,
    passwordValid,
  } = useForm(formValues, formValidations);

  const onSubmit = (event) => {
    event.preventDefault();
    setFormSubmitted(true);
    console.log(formState);
  };

  return (
    <AuthLayout formTitle="Register">
      <form onSubmit={onSubmit}>
        <Grid container>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              label="Nombre"
              type="text"
              placeholder="Nombre completo"
              fullWidth
              name="displayName"
              value={displayName}
              onChange={onInputChange}
              error={!!displayNameValid && formSubmitted}
              helperText={displayNameValid}
            />
          </Grid>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              label="Correo"
              type="email"
              placeholder="correo@google.com"
              fullWidth
              name="email"
              value={email}
              onChange={onInputChange}
              error={!!emailValid && formSubmitted}
              helperText={emailValid}
            />
          </Grid>
          <Grid item xs={12} sx={{ mt: 2 }}>
            <TextField
              label="Contraseña"
              type="password"
              placeholder="contraseña"
              fullWidth
              name="password"
              value={password}
              onChange={onInputChange}
              error={!!passwordValid && formSubmitted}
              helperText={passwordValid}
            />
          </Grid>
        </Grid>

        <Grid container spacing={2} sx={{ mb: 2, mt: 1 }}>
          <Grid item xs={12} sm={12}>
            <Button
              type="submit"
              onSubmit={onSubmit}
              variant="contained"
              fullWidth
            >
              Register
            </Button>
          </Grid>
        </Grid>

        <Grid container direction="row" justifyContent="end">
          <Typography sx={{ mr: 1 }}>¿Ya tienes una cuenta?</Typography>
          <Link component={Routerlink} to="/auth/login">
            Iniciar Sesion
          </Link>
        </Grid>
      </form>
    </AuthLayout>
  );
};
