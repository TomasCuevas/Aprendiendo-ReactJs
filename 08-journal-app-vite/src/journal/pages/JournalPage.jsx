import { Typography } from '@mui/material'
import { JournalLayout } from '../layout/JournalLayout'

export const JournalPage = () => {

  return (
    <JournalLayout>
      <Typography>Voluptate excepteur officia elit tempor in reprehenderit. Exercitation eu esse voluptate qui elit laborum fugiat minim officia magna ad ullamco. Sit et irure dolore anim commodo eu dolore ipsum ea.</Typography>
    </JournalLayout>
  )
}