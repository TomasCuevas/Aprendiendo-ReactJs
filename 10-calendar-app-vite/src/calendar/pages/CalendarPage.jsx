import { Calendar } from 'react-big-calendar';
import { addHours } from 'date-fns';
import 'react-big-calendar/lib/css/react-big-calendar.css';

import { Navbar, CalendarEventBox } from '../';

import { localizer, getMessagesES } from '../../helpers';

const events = [
  {
    title: 'CumpleaÃ±os del Jefe',
    notes: 'Hay que comprar pastel',
    start: new Date(),
    end: addHours(new Date(), 2),
    user: {
      id: '123',
      name: 'Tomas',
    },
  },
];

export const CalendarPage = () => {
  const eventStylesGetter = (event, start, end, isSelected) => {
    const style = {
      backgroundColor: '#347CF7',
      borderRadius: '0px',
      opacity: 0.8,
      color: 'white',
    };

    return {
      style,
    };
  };

  return (
    <>
      <Navbar />
      <Calendar
        culture="es"
        localizer={localizer}
        events={events}
        startAccessor="start"
        endAccessor="end"
        style={{ height: 'calc(100vh - 80px)' }}
        messages={getMessagesES()}
        eventPropGetter={eventStylesGetter}
        components={{
          event: CalendarEventBox,
        }}
      />
    </>
  );
};
