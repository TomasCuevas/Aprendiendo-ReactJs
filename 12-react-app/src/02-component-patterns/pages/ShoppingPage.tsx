import { ProductButtons, ProductCard, ProductImage, ProductTitle } from '../components/ProductCard';

const product = {
  id: '1',
  title: 'Coffee Mug - Card',
  img: './coffee-mug.png',
};

export const ShoppingPage = () => {
  return (
    <div>
      <h1>Shopping Store</h1>
      <hr />

      <div style={{ display: 'flex', flexDirection: 'row', flexWrap: 'wrap' }}>
        <ProductCard product={product}>
          <ProductImage {...product} />
          <ProductTitle {...product} />
          {/* <ProductButtons {...product} /> */}
        </ProductCard>

        <ProductCard product={product}>
          <ProductCard.Image {...product} />
          <ProductCard.Title {...product} />
          {/* <ProductButtons {...product} /> */}
        </ProductCard>
      </div>
    </div>
  );
};
