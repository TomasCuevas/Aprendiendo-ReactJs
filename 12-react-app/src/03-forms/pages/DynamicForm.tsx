import { Formik, Form } from 'formik';
import * as Yup from 'yup';

import { MySelect, MyTextInput } from '../components';
import formJson from '../data/custom-form.json';

const initialValues: { [key: string]: any } = {};
const requiredField: { [key: string]: any } = {};

for (const input of formJson) {
  initialValues[input.name] = input.value;

  if (!input.validations) continue;

  let schema = Yup.string();

  for (const rule of input.validations) {
    if (rule.type === 'required')
      schema = schema.required(`${input.label} is a required field`);
    if (rule.type === 'min')
      schema = schema.min(
        (rule as any).value,
        `${input.label} must have at least ${rule.value} characters `,
      );
    if (rule.type === 'email')
      schema = schema.email(`${input.label}, must be a valid email.`);
  }

  requiredField[input.name] = schema;
}

const validationSchema = Yup.object({ ...requiredField });

export const DynamicForm = () => {
  return (
    <div>
      <h1>DynamicForm</h1>
      <Formik
        initialValues={initialValues}
        onSubmit={(values) => console.log(values)}
        validationSchema={validationSchema}
      >
        {(formik) => (
          <Form noValidate>
            {formJson.map(({ type, label, name, placeholder, options }) => {
              if (
                type === 'email' ||
                type === 'password' ||
                type === 'text' ||
                type === 'input'
              ) {
                return (
                  <MyTextInput
                    key={name}
                    label={label}
                    name={name}
                    type={type as any}
                    placeholder={placeholder}
                  />
                );
              }

              if (type === 'select') {
                return (
                  <MySelect key={name} label={label} name={name}>
                    <option value="">Select an option</option>
                    {options?.map(({ id, label }) => (
                      <option key={id} value={id}>
                        {label}
                      </option>
                    ))}
                  </MySelect>
                );
              }

              return <span>Type: {type} no es soportado</span>;
            })}

            <button type="submit">Register</button>
          </Form>
        )}
      </Formik>
    </div>
  );
};
