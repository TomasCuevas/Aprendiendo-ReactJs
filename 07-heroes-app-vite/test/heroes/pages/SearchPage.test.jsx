import { render, screen } from "@testing-library/react"
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from "../../../src/heroes/pages/SearchPage";

describe('Pruebas en el <SearchPage />', () => {

  test('debe de mostrarse correctamente con valores por defecto', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect( container ).toMatchSnapshot();
  });

  test('debe de mostrar a Batman y el input con el valor del queryString', () => {
    const search = 'batman'
    
    const { container } = render(
      <MemoryRouter initialEntries={[`/search?q=${search}`]}>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByLabelText( 'search-input' );
    expect( input.value ).toBe( search )

    const img = screen.getByRole('img');
    expect( img.src ).toContain( '/assets/heroes/dc-batman.jpg' )
  });

})