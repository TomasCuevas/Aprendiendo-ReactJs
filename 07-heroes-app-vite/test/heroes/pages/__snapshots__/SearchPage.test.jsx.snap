// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <SearchPage /> debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5 mt-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          aria-label="search-input"
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          aria-label="search-button"
          class="btn btn-outline-primary mt-2"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7 mt-5"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
